main/src/ca/uwaterloo/flix/language/phase/jvm/BackendObjType.scala

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2021 Jonathan Lindegaard Starup
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ca.uwaterloo.flix.language.phase.jvm
+
+import ca.uwaterloo.flix.language.ast.Symbol
+import ca.uwaterloo.flix.language.phase.jvm.JvmName.{Delimiter, DevFlixRuntime, JavaLang, RootPackage}
+
+/**
+  * Represents all Flix types that are objects on the JVM (array is an exception).
+  */
+sealed trait BackendObjType {
+  /**
+    * Constructs a concatenated list of erased strings delimited with `JvmName.Delimiter`.
+    */
+  private def erasedListOfTypes(ts: List[BackendType]): String =
+    ts.map(e => e.toErased.toString).mkString(Delimiter)
+
+  val jvmName: JvmName = this match {
```

Comment:
Add comments for every method. 

Public methods should come before private methods.